client/src/components/mapping/TabColumnPicking/DynamicColumnPicker/index.tsx

```diff
@@ -35,31 +38,42 @@ const DynamicColumnPicker = ({ attribute, schema, source }: Props) => {
   const dispatch = useDispatch();
   const toaster = useSelector((state: IReduxStore) => state.toaster);
 
-  const { resource } = useSelector((state: IReduxStore) => state.selectedNode);
-  const path = attribute.path;
-
+  const { resource, selectedInputGroup } = useSelector(
+    (state: IReduxStore) => state.selectedNode
+  );
   const attributesForResource = useSelector(
     (state: IReduxStore) => state.resourceInputs.attributesMap
   );
+
+  const path = attribute.path;
   let attributeId = attributesForResource[path]
     ? attributesForResource[path].id
     : null;
+  let inputGroupId =
```

Comment:
please explain wtf is going on here, this is too hard to read without a pint of coffee